### ISSUE TITLE
LoadError: cannot load such file -- htauth

### ISSUE THREAD

**Issue by fletchowns** [sous-chefs/htpasswd#33]
Getting a converge error on debian-7.8 with htpasswd 0.3.0 and default attributes. I recently upgraded to Chef 13 so I'm thinking that may be the culprit.

```
LoadError: cannot load such file -- htauth
/opt/chef/embedded/lib/ruby/2.4.0/rubygems/core_ext/kernel_require.rb:55:in `require'
/opt/chef/embedded/lib/ruby/2.4.0/rubygems/core_ext/kernel_require.rb:55:in `require'
/tmp/kitchen/cache/cookbooks/htpasswd/providers/default.rb:60:in `initialize'
/opt/chef/embedded/lib/ruby/gems/2.4.0/gems/chef-13.3.42/lib/chef/resource.rb:1399:in `new'
/opt/chef/embedded/lib/ruby/gems/2.4.0/gems/chef-13.3.42/lib/chef/resource.rb:1399:in `provider_for_action'
/opt/chef/embedded/lib/ruby/gems/2.4.0/gems/chef-13.3.42/lib/chef/resource.rb:591:in `run_action'
/opt/chef/embedded/lib/ruby/gems/2.4.0/gems/chef-13.3.42/lib/chef/runner.rb:70:in `run_action'
/opt/chef/embedded/lib/ruby/gems/2.4.0/gems/chef-13.3.42/lib/chef/runner.rb:98:in `block (2 levels) in converge'
/opt/chef/embedded/lib/ruby/gems/2.4.0/gems/chef-13.3.42/lib/chef/runner.rb:98:in `each'
/opt/chef/embedded/lib/ruby/gems/2.4.0/gems/chef-13.3.42/lib/chef/runner.rb:98:in `block in converge'
/opt/chef/embedded/lib/ruby/gems/2.4.0/gems/chef-13.3.42/lib/chef/resource_collection/resource_list.rb:94:in `block in execute_each_resource'
/opt/chef/embedded/lib/ruby/gems/2.4.0/gems/chef-13.3.42/lib/chef/resource_collection/stepable_iterator.rb:114:in `call_iterator_block'
/opt/chef/embedded/lib/ruby/gems/2.4.0/gems/chef-13.3.42/lib/chef/resource_collection/stepable_iterator.rb:85:in `step'
/opt/chef/embedded/lib/ruby/gems/2.4.0/gems/chef-13.3.42/lib/chef/resource_collection/stepable_iterator.rb:103:in `iterate'
/opt/chef/embedded/lib/ruby/gems/2.4.0/gems/chef-13.3.42/lib/chef/resource_collection/stepable_iterator.rb:55:in `each_with_index'
/opt/chef/embedded/lib/ruby/gems/2.4.0/gems/chef-13.3.42/lib/chef/resource_collection/resource_list.rb:92:in `execute_each_resource'
/opt/chef/embedded/lib/ruby/gems/2.4.0/gems/chef-13.3.42/lib/chef/runner.rb:97:in `converge'
/opt/chef/embedded/lib/ruby/gems/2.4.0/gems/chef-13.3.42/lib/chef/client.rb:715:in `block in converge'
/opt/chef/embedded/lib/ruby/gems/2.4.0/gems/chef-13.3.42/lib/chef/client.rb:710:in `catch'
/opt/chef/embedded/lib/ruby/gems/2.4.0/gems/chef-13.3.42/lib/chef/client.rb:710:in `converge'
/opt/chef/embedded/lib/ruby/gems/2.4.0/gems/chef-13.3.42/lib/chef/client.rb:749:in `converge_and_save'
/opt/chef/embedded/lib/ruby/gems/2.4.0/gems/chef-13.3.42/lib/chef/client.rb:286:in `run'
/opt/chef/embedded/lib/ruby/gems/2.4.0/gems/chef-13.3.42/lib/chef/application.rb:291:in `block in fork_chef_client'
/opt/chef/embedded/lib/ruby/gems/2.4.0/gems/chef-13.3.42/lib/chef/application.rb:279:in `fork'
/opt/chef/embedded/lib/ruby/gems/2.4.0/gems/chef-13.3.42/lib/chef/application.rb:279:in `fork_chef_client'
/opt/chef/embedded/lib/ruby/gems/2.4.0/gems/chef-13.3.42/lib/chef/application.rb:244:in `block in run_chef_client'
/opt/chef/embedded/lib/ruby/gems/2.4.0/gems/chef-13.3.42/lib/chef/local_mode.rb:44:in `with_server_connectivity'
/opt/chef/embedded/lib/ruby/gems/2.4.0/gems/chef-13.3.42/lib/chef/application.rb:232:in `run_chef_client'
/opt/chef/embedded/lib/ruby/gems/2.4.0/gems/chef-13.3.42/lib/chef/application/client.rb:469:in `sleep_then_run_chef_client'
/opt/chef/embedded/lib/ruby/gems/2.4.0/gems/chef-13.3.42/lib/chef/application/client.rb:458:in `block in interval_run_chef_client'
/opt/chef/embedded/lib/ruby/gems/2.4.0/gems/chef-13.3.42/lib/chef/application/client.rb:457:in `loop'
/opt/chef/embedded/lib/ruby/gems/2.4.0/gems/chef-13.3.42/lib/chef/application/client.rb:457:in `interval_run_chef_client'
/opt/chef/embedded/lib/ruby/gems/2.4.0/gems/chef-13.3.42/lib/chef/application/client.rb:441:in `run_application'
/opt/chef/embedded/lib/ruby/gems/2.4.0/gems/chef-13.3.42/lib/chef/application.rb:59:in `run'
/opt/chef/embedded/lib/ruby/gems/2.4.0/gems/chef-13.3.42/bin/chef-client:26:in `<top (required)>'
/opt/chef/bin/chef-client:58:in `load'
/opt/chef/bin/chef-client:58:in `<main>'
```

**Comment by guilhem** [comment]
I don't test it on chef 13 :/

Can you try to change `gem 'htauth' if respond_to?(:gem)` to `gem 'htauth'` in `metadata.rb`?

**Comment by fletchowns** [comment]
@guilhem that seemed to fix it! Thanks for the suggestion, and thanks for the handy cookbook!!

**Comment by quulah** [comment]
I'm getting the same error on Chef 12.21.12. Same fix applies.

**Comment by EugenMayer** [comment]
same here

**Comment by jpmat296** [comment]
I've got the same issue. Root cause seems bad metadata.json published to Chef supermarket. For the same reason, "issue_url" is not visible in supermarket:

<img width="1150" alt="image" src="https://user-images.githubusercontent.com/12024504/57129743-a6f3de00-6d97-11e9-91ab-05515a048931.png">

A new publication to supermarket should fix the issue.

**Comment by jpmat296** [comment]
A simple workaround is to add git source in Berksfile in your dependant cookbook :
`cookbook 'htpasswd', git: 'https://github.com/redguide/htpasswd.git'`

By this way, metadata.json file is generated and contains :
```
  "gems": [
    [
      "htauth"
    ]
  ]
```

**Comment by EugenMayer** [comment]
Last commit - 2 years ago. so if you care, fork an re-release. There will be nothing happening here anyway :)

**Comment by guilhem** [comment]
We are not maintaining it anymore.
But as said a number of time, we will be happy to add new maintainers :)

As I'm not doing any chef anymore, I don't want to push something I can't test... even if old version have some "problem". :(

**Comment by EugenMayer** [comment]
@guilhem no offense, that happense, it's OSS. You should archive this repo probably to make it crystal clear, that there is no development happening and that you search for a new maintainer ( or a good fork ).

If there is enough interest, someone picks it up - or not.

Thank you for your time anyway :)

**Comment by guilhem** [comment]
@EugenMayer oh sorry if my tone look like I was offense, I wrote it maybe too fast before going in weekend ^^.

Yes that's maybe what should happen... but as you see, I created _redguide_ to be an "open" organization for anyone to jump-in. And now I can't archive it only by myself :)

**Comment by ramereth** [comment]
This has been addressed in #41
